src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs

```diff
@@ -0,0 +1,152 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class PseudoCustomAttributeTests
+    {
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public static void IsDefined(object target, Attribute attribute)
+        {
+            switch (target)
+            {
+            case Type type:
+                Assert.True(Attribute.IsDefined(type, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(type, attribute.GetType(), true));
+                break;
+            case MemberInfo memberInfo:
+                Assert.True(Attribute.IsDefined(memberInfo, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(memberInfo, attribute.GetType(), true));
+                break;
+            case ParameterInfo parameterInfo:
+                Assert.True(Attribute.IsDefined(parameterInfo, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(parameterInfo, attribute.GetType(), true));
```

Comment:
Do we have any tests validating that IsDefined returns false when it's supposed to?

Reply:
Added.